assets/src/edit-story/components/panels/stylePreset/panel.js

```diff
@@ -0,0 +1,203 @@
+import { getShapePresets, getTextPresets } from './utils';
+import PresetsHeader from './header';
+import Presets from './presets';
+import Resize from './resize';
+
+function StylePresetPanel() {
+  const {
+    state: {
+      selectedElementIds,
+      selectedElements,
+      story: { stylePresets },
+    },
+    actions: { updateStory, updateElementsById },
+  } = useStory();
+
+  const { fillColors, textColors, textStyles } = stylePresets;
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  const isType = (elType) => {
+    return (
+      selectedElements.length > 0 &&
+      selectedElements.every(({ type }) => elType === type)
+    );
+  };
+
+  const isText = isType('text');
+  const isShape = isType('shape');
+
+  const handleDeletePreset = useCallback(
+    (toDelete) => {
+      updateStory({
+        properties: {
+          stylePresets: {
+            textStyles: textStyles.filter((style) => style !== toDelete),
+            fillColors: isText
+              ? fillColors
+              : fillColors.filter((color) => color !== toDelete),
+            textColors: !isText
+              ? textColors
+              : textColors.filter((color) => color !== toDelete),
+          },
+        },
+      });
+    },
+    [textStyles, fillColors, isText, textColors, updateStory]
+  );
+
+  const handleAddColorPreset = useCallback(
+    (evt) => {
+      evt.stopPropagation();
+      let addedPresets = {
+        fillColors: [],
+        textColors: [],
+        textStyles: [],
+      };
+      if (isText) {
+        addedPresets = {
+          ...addedPresets,
+          ...getTextPresets(selectedElements, stylePresets),
+        };
+      } else {
+        // Currently, shape only supports fillColors.
+        addedPresets = {
+          ...addedPresets,
+          ...getShapePresets(selectedElements, stylePresets),
+        };
+      }
+      if (
+        addedPresets.fillColors?.length > 0 ||
+        addedPresets.textColors?.length > 0 ||
+        addedPresets.textStyles?.length > 0
+      ) {
+        updateStory({
+          properties: {
+            stylePresets: {
+              textStyles: [...textStyles, ...addedPresets.textStyles],
+              fillColors: [...fillColors, ...addedPresets.fillColors],
+              textColors: [...textColors, ...addedPresets.textColors],
+            },
+          },
+        });
+      }
+    },
+    [
+      fillColors,
+      textStyles,
+      textColors,
+      isText,
+      selectedElements,
+      updateStory,
+      stylePresets,
+    ]
+  );
+
+  const handleApplyPreset = useCallback(
+    (preset) => {
+      if (isText) {
+        // @todo Determine this in a better way.
+        // Only style presets have background text mode set.
+        const isStylePreset = preset.backgroundTextMode !== undefined;
+        updateElementsById({
+          elementIds: selectedElementIds,
+          properties: isStylePreset ? { ...preset } : { color: preset },
```

Comment:
/cc @barklund 

So, this will break completely with new text styling. Perhaps we should just proceed here and fix once that lands? I don't think it'd be a major update at that point. Instead of updating a `color`, we would be updating `content` field.

Reply:
Note that the text presets currently already are using `color` for text which Morten will have to re-work in the new text editing PR anyway. Probably the change coming from this PR won't be that much different to rework compared to what already exists for Color Presets currently, meaning there shouldn't be a huge difference in terms of which lands first from Morten's PR's perspective.

An alternative would be leaving the color out of the Text editing PR for now and handling it together with presets separately to reduce the size of the text editing PR. There might be some UX questions to resolve anyway, e.g.
If the user selects part of the text and clicks on a color preset, should the color be applied to the selection only? If there are multiple colors in an element and the user clicks "Add preset", should a preset be added for each color? Or should a preset be added only when color is applied to the full Text?

That's probably a better discussion under #1323, however, just making a suggestion that perhaps color might be something to handle separately together with Style Presets in a separate PR. Thoughts?

cc @barklund

Reply:
I'd prefer this PR to be merged first and do the fixes in #1323 (with your help, @miina, if I don't get all the aspects of color presets correct)

Reply:
IMHO, at this point it will be easier to pursue separately and fix upstream, since both are rather new features.